### PR TITLE
Filled button's icon on hover

### LINKED ISSUE #20179: vNext Button Subtle and Transparent Variant, Icon should be Filled for Hover/Pressed/Toggle checked on states
**Current**
Hover: 
image

Toggle checked on:
image

Environment
https://fluentuipr.z22.web.core.windows.net/heads/master/react-components/storybook/index.html?path=/story/components-togglebutton--toggle-button-playground&knob-Theme=[object%20Object]

**Change Request**
image

On Hover, Pressed, and Toggle checked on states, the icon should use the filled version. ([Referencing this for icon naming convention](https://github.com/microsoft/fluentui-system-icons/blob/master/icons.md)) 

At Rest:  (Icon name)20Regular
For Hover/Pressed/Toggle Checked on: (Icon name) 20Filled

So i.e. 
At Rest: chatVideo20Regular
For Hover/Pressed/Toggle Checked on: chatVideo20Filled

### LINKED ISSUE #22464: [Bug]: Button V9 Teams theme: High contrast text color is white, should be black to meet accessibility requirements
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
v9
```

### Are you reporting Accessibility issue?

no

### Reproduction

https://6002298f95a00c00213f4d55-qpafvorpoj.chromatic.com/?path=/docs/components-button--default

## Actual Behavior
color is white

## Expected Behavior
color should be black

### Logs

_No response_

### Requested priority

High

### Products/sites affected

_No response_

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by annabratseiko**
## Current Behavior

`Button's` icon is not filled on hover or on checked for `ToggleButtons`.

## New Behavior

Added ability to specify via `bundleIcon` icons that should be filled on hover.
image

## Related Issue(s)

Fixes #20179

### CODE REVIEW

File: packages/react-button/src/components/Button/stories/ButtonIcon.stories.tsx [hunk 1]
```diff
@@ -1,19 +1,22 @@
...
-    <Button icon={<CalendarMonthRegular />} iconPosition="after">
-      With calendar icon after contents
-    </Button>
-    <Tooltip content="With calendar icon only" relationship="label">
-      <Button icon={<CalendarMonthRegular />} />
-    </Tooltip>
-  </>
-);
+export const Icon = () => {
+  const CalendarMonth = bundleIcon(CalendarMonthFilled, CalendarMonthRegular);
```
Comment (layershifter) [inline packages/react-button/src/components/Button/stories/ButtonIcon.stories.tsx:7]: You can call `bundleIcon` in root scope:

```suggestion
const CalendarMonth = bundleIcon(CalendarMonthFilled, CalendarMonthRegular);

export const Icon = () => {
```

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                          |    Baseline (minified/GZIP) |                          PR |                                                                                                                                                                                                   Change |
| :--------------------------------------------------------------------------------------------------------- | --------------------------: | --------------------------: | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/Button.fixture.js'>Button</abbr>                 |  `28.083 kB`<br />`8.09 kB` | `28.359 kB`<br />`8.196 kB` | `276 B` <br />`106 B`  |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/CompoundButton.fixture.js'>CompoundButton</abbr> | `33.578 kB`<br />`9.126 kB` | `33.854 kB`<br />`9.231 kB` | `276 B` <br />`105 B`  |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/MenuButton.fixture.js'>MenuButton</abbr>         |  `29.85 kB`<br />`8.684 kB` | `30.126 kB`<br />`8.789 kB` | `276 B` <br />`105 B`  |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/SplitButton.fixture.js'>SplitButton</abbr>       | `36.322 kB`<br />`9.878 kB` | `36.598 kB`<br />`9.981 kB` | `276 B` <br />`103 B`  |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/ToggleButton.fixture.js'>ToggleButton</abbr>     | `37.465 kB`<br />`8.717 kB` | `37.741 kB`<br />`8.821 kB` | `276 B` <br />`104 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                  |         Size (minified/GZIP) |
| -------------------------------------------------------------------------------------------------- | ---------------------------: |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/Card.All.fixture.js'>Card - All</abbr>     | `57.884 kB`<br />`16.524 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/Card.fixture.js'>Card</abbr>               | `53.169 kB`<br />`15.238 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardFooter.fixture.js'>CardFooter</abbr>   |   `7.756 kB`<br />`3.299 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardHeader.fixture.js'>CardHeader</abbr>   |   `9.321 kB`<br />`3.816 kB` |
| <samp>react-card</samp> <br /> <abbr title='bundle-size/CardPreview.fixture.js'>CardPreview</abbr> |   `7.728 kB`<br />`3.323 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/f3c6462af3efa4f8cbe32d289e8e3d4a450b8a3b'>f3c6462af3efa4f8cbe32d289e8e3d4a450b8a3b</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22587/perf-test-react-components/Avatar-mount_base.html">1245</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22587/perf-test-react-components/Avatar-mount.html">1223</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22587/perf-test-react-components/Button-mount_base.html">729</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22587/perf-test-react-components/Button-mount.html">745</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22587/perf-test-react-components/FluentProvider-mount_base.html">2180</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22587/perf-test-react-components/FluentProvider-mount.html">2138</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22587/perf-test-react-components/FluentProviderWithTheme-mount_base.html">379</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22587/perf-test-react-components/FluentProviderWithTheme-mount.html">373</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22587/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">353</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22587/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">345</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22587/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">448</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22587/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">438</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22587/perf-test-react-components/MakeStyles-mount_base.html">1935</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22587/perf-test-react-components/MakeStyles-mount.html">1989</a></td>
            <td>50000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Review by spmonahan — APPROVED** [review]

File: change/@fluentui-react-button-632b8f65-cf5d-4f77-a61a-3cee616e7d44.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "none",
+  "comment": "Merge branch 'master' into annkelle/button_hover_icon",
```
Comment (khmakoto) [inline change/@fluentui-react-button-632b8f65-cf5d-4f77-a61a-3cee616e7d44.json:3]:
```suggestion
  "comment": "Button: Use filled icons on hover.",
```

File: packages/react-button/src/components/Button/stories/ButtonIcon.stories.tsx [hunk 3]
```diff
@@ -1,16 +1,18 @@
 import * as React from 'react';
-import { CalendarMonthRegular } from '@fluentui/react-icons';
+import { CalendarMonthRegular, CalendarMonthFilled, bundleIcon } from '@fluentui/react-icons';
```
Comment (khmakoto) [inline packages/react-button/src/components/Button/stories/ButtonIcon.stories.tsx:2]: _nit:_ Order alphabetically

```suggestion
import { bundleIcon, CalendarMonthRegular, CalendarMonthFilled } from '@fluentui/react-icons';
```

**Review by khmakoto — APPROVED** [review]

**Pull request merged**